### PR TITLE
[kots]: automatically enable shiftfs support if cluster supports it

### LINKED ISSUE #8883: Add check to see if should use shiftfs or fuse
> See [internal discussion](https://gitpod.slack.com/archives/C02F19UUW6S/p1647600494757959)

By default, we use `fuse` over `shiftfs` for the file systems on the installation. ShiftFS has some performance advantages, but requires messing with the kernel module - Fuse is therefore easier to get set up with.

Add a test in the `gitpod-installer-job.yaml` to decide if `shiftfs` can be used and to fallback to `fuse` if not.

## Suggested approach

1. Run the [shiftfs-module-loader](https://github.com/gitpod-io/gitpod/tree/main/components/ws-daemon/shiftfs-module-loader).
2. If exits with code 0, add `shiftfs` in the `yq` amendment section
3. Else, do nothing (remaining with `fuse`)

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

This is a PR with a proposed change to perform a check to see if we can enable ShiftFS support. This adds the `shiftfs-module-loader` image as a job and the Installer checks whether it's completed. It only switches to ShiftFS if it's passed, otherwise it sticks with Fuse.

I think the first question that needs to be answered - is this a sensible way of performing this check? I've tried it in GKE (with ShiftFS available) and K3s (without ShiftFS available) and I get the desired results.

## Related Issue(s)

Fixes #8883

## How to test

Install Gitpod on

## Release Notes

```release-note
[kots]: automatically enable shiftfs support if cluster supports it
```

### CODE REVIEW

**Comment by roboquat** [comment]
@MrSimonEmms: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by MrSimonEmms** [comment]
werft run github -a no-preview publish-to-kots -f

File: install/kots/manifests/gitpod-shiftfs-module-loader.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    metadata:
+      labels:
+        app: gitpod
+        component: shiftfs-module-loader
+    spec:
+      serviceAccountName: installer
+      restartPolicy: Never
+      containers:
+        - name: shiftfs-module-loader
+          image: eu.gcr.io/gitpod-core-dev/build/shiftfs-module-loader:commit-ab235e8bc00f2c1ac70232cea17e5df9a9d262fa
```
Comment (MrSimonEmms) [inline install/kots/manifests/gitpod-shiftfs-module-loader.yaml:26]: This is the current valid tag. This is generated by the Werft job

Comment (corneliusludmann) [reply install/kots/manifests/gitpod-shiftfs-module-loader.yaml:27]: NIT: Would love to see this as a comment in the YAML file as well! :innocent: :stuck_out_tongue_closed_eyes:

**Review by corneliusludmann — APPROVED** [review]
Great improvement!

Holding in case you want to accept my awesome, very important NIT comment to make this pull request even better. :see_no_evil: No hard feelings if not. :rofl: 
/hold

**Comment by MrSimonEmms** [comment]
@corneliusludmann ooooh, you're a hard taskmaster

image

File: install/kots/manifests/gitpod-shiftfs-module-loader.yaml (hunk 1 above)
Comment (MrSimonEmms) [reply install/kots/manifests/gitpod-shiftfs-module-loader.yaml:28]: Nit resolved, hopefully to your satisfaction

**Pull request merged**